solidity/contracts/MailboxV2.sol

```diff
@@ -0,0 +1,224 @@
+
+contract MailboxV2 is
+    IMailboxV2,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    Versioned
+{
+    // ============ Libraries ============
+
+    using MerkleLib for MerkleLib.Tree;
+    using MessageV2 for bytes;
+    using TypeCasts for bytes32;
+    using TypeCasts for address;
+
+    // ============ Constants ============
+
+    // Maximum bytes per message = 2 KiB (somewhat arbitrarily set to begin)
+    uint256 public constant MAX_MESSAGE_BODY_BYTES = 2 * 2**10;
+    // Domain of chain on which the contract is deployed
+    uint32 public immutable localDomain;
+
+    // ============ Public Storage ============
+
+    // The default ISM, used if the recipient fails to specify one.
+    IInterchainSecurityModule public defaultModule;
+    // An incremental merkle tree used to store outbound message IDs.
+    MerkleLib.Tree public tree;
+    // Mapping of message ID to whether or not that message has been delivered.
+    mapping(bytes32 => bool) public delivered;
+
+    // ============ Upgrade Gap ============
+
+    // gap for upgrade safety
+    uint256[47] private __GAP;
+
+    // ============ Events ============
+
+    /**
+     * @notice Emitted when the default ISM is updated
+     * @param module The new default ISM
+     */
+    event DefaultModuleSet(address indexed module);
+
+    /**
+     * @notice Emitted when a new message is dispatched via Hyperlane
+     * @param messageId The unique message identifier
+     * @param message Raw bytes of message
+     */
+    event Dispatch(bytes32 indexed messageId, bytes message);
+
+    /**
+     * @notice Emitted when a Hyperlane message is delivered
+     * @param messageId The unique message identifier
+     */
+    event Process(bytes32 indexed messageId);
+
+    // ============ Constructor ============
+
+    // solhint-disable-next-line no-empty-blocks
+    constructor(uint32 _localDomain) {
+        localDomain = _localDomain;
+    }
+
+    // ============ Initializer ============
+
+    function initialize(address _defaultModule) external initializer {
+        __ReentrancyGuard_init();
+        __Ownable_init();
+        _setDefaultModule(_defaultModule);
+    }
+
+    // ============ External Functions ============
+
+    /**
+     * @notice Sets the default ISM for the Mailbox.
+     * @param _module The new default ISM. Must be a contract.
+     */
+    function setDefaultModule(address _module) external onlyOwner {
+        _setDefaultModule(_module);
+    }
+
+    /**
+     * @notice Dispatches a message to the destination domain & recipient.
+     * @param _destinationDomain Domain of destination chain
+     * @param _recipientAddress Address of recipient on destination chain as bytes32
+     * @param _messageBody Raw bytes content of message body
+     * @return The message ID inserted into the Mailbox's merkle tree
+     */
+    function dispatch(
```

Comment:
Is it? Take a look at `Mailbox.sol`, I think you'll see basically zero overlap between the two. V1 is just some functions around the validator manager, which ofc are no longer relevant.

 I'm not sure that naming this file `Mailbox.sol` would lead to a useful diff.